ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -194,6 +207,50 @@ def _llmobs_set_meta_tags_from_chain(
             except TypeError:
                 log.warning("Failed to serialize chain output data to JSON")
 
+    def _llmobs_set_meta_tags_from_embedding(
+            self,
+            span: Span,
+            input_texts: Union[str, List[str]],
+            output_embedding: Union[List[float], List[List[float]], None],
+            error: bool = False,
+            is_workflow: bool = False,
+    ) -> None:
+        span.set_tag_str(SPAN_KIND, "workflow" if is_workflow else "embedding")
+        span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
+        span.set_tag_str(MODEL_PROVIDER, span.get_tag(PROVIDER) or "")
+
+        input_tag_key = INPUT_VALUE if is_workflow else INPUT_DOCUMENTS
+        output_tag_key = OUTPUT_VALUE
+
+        if (input_texts is not None
+                and (isinstance(input_texts, str) or all(isinstance(text, str) for text in input_texts))):
+            if is_workflow:
+                try:
+                    formatted_inputs = self.format_io(input_texts)
+                    if isinstance(input_texts, str):
+                        span.set_tag_str(input_tag_key, formatted_inputs)
+                    else:
+                        span.set_tag_str(input_tag_key, json.dumps(self.format_io(input_texts)))
+                except TypeError:
+                    log.warning("Failed to serialize embedding input data to JSON")
+            else:
+                if isinstance(input_texts, str):
+                    span.set_tag_str(input_tag_key, input_texts)
+                elif (isinstance(input_texts, list)
+                      and all(isinstance(text, str) for text in input_texts) and not is_workflow):
+                    json_documents_list = json.dumps(input_texts)
+                    span.set_tag_str(input_tag_key, '{"documents":%s}' % json_documents_list)
```

Comment:
Two suggestions here:
1. Let's avoid double nesting conditionals if possible 
```suggestion
            elif isinstance(input_texts, str):
                    span.set_tag_str(input_tag_key, input_texts)
            elif (isinstance(input_texts, list)
                      and all(isinstance(text, str) for text in input_texts) and not is_workflow):
                    json_documents_list = json.dumps(input_texts)
                    span.set_tag_str(input_tag_key, '{"documents":%s}' % json_documents_list)
```
2. Can we construct the `Document` objects similar to how we do it in the OpenAI integration? That might be a simpler/more readable approach (might be worth to add a try/catch similar to how you did in line 234)